src/xunit.analyzers/InlineDataMustMatchTheoryParameters.cs

```diff
@@ -78,52 +86,77 @@ public override void Initialize(AnalysisContext context)
                                 builder.ToImmutable()));
                         }
 
-                        for (int i = 0; i < Math.Min(values.Length, method.Parameters.Length); i++)
+                        int valueIdx = 0, paramIdx = 0;
+                        for (; valueIdx < values.Length && paramIdx < method.Parameters.Length; valueIdx++)
                         {
-                            var parameter = method.Parameters[i];
+                            var parameter = method.Parameters[paramIdx];
+
+                            if (xunitSupportsParameterArrays
+                                && parameter.IsParams
+                                && parameter is IArrayTypeSymbol arrayParameter
+                                && arrayParameter.ElementType == compilation.ObjectType)
+                            {
+                                // Everything is assignable to 'params object[]' so move one
+                                continue;
```

Comment:
I think in this case it could actually be changed into
```
valueIdx = values.Lenght; // Make sure we don't validate remaining values
break;
```
Since the remaining values will be assignable.